patterns/singleton/Singleton-lazy.cs

```diff
@@ -0,0 +1,108 @@
+    private static readonly Lazy<Student> instance = new Lazy<Student>(() => new Student("Default",0,"None"));
+    
+    // Public static property to access the singleton instance
+    public static Student Instance
+    {
+        get
+        {
+            return instance.Value;
+        }
+    }
+
+    // Constructor
+    private Student(string name, int age, string grade)  // Making private constructor to prevent instantiation from outside
+    {
+        Name = name;
+        Age = age;
+        Grade = grade;
+    }
+
+    // Method to update student details
+    public void UpdateDetails(string Name, int Age, string Grade)
+    {
+        this.Name = Name;
+        this.Age = Age;
+        this.Grade = Grade;
+    }
+
+    // Method to display student details
+    public void DisplayDetails()
+    {
+        Console.WriteLine($"Name: {Name}");
+        Console.WriteLine($"Age: {Age}");
+        Console.WriteLine($"Grade: {Grade}");
+    }
+}
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        // Accessing the singleton instance
+        Student student = Student.Instance;
+
+        // Display original details
+        Console.WriteLine("Original Details:");
+        student.DisplayDetails();
+
+        // Update student details
+        student.UpdateDetails("Ravi", 30, "B");
+
+        // Display updated details
+        Console.WriteLine("\nUpdated Details:");
+        student.DisplayDetails();
+
+        /// <summary>
+        ///  If developer try to create the new object of student class, he will get error at the compile time
+        ///  Comile time error: Student.Student(string, int, string)' is inaccessible due to its protection level
+        /// </summary>
+        
+        // <example> 
+
+        //Student student1 = new Student("",0,""); 
+        //student1.UpdateDetails("R",12,"C");
+        
+        // </example>
+
+        /// <summary>
+        /// Developer can check the using the bool if the instance is exist before operating the operation 
+        ///  Output if Instace exist: Is the instance created? True
+        /// </summary>
+        
+        //  <example>
+
+        bool isInstanceCreated = Student.Instance != null; //Checking if the instance has been created yet
+        Console.WriteLine($"Is the instance created? {isInstanceCreated}"); 
+
+        // </example>
+
+
+        /// <summary>
+        /// Developer can Create the another instance and can access the Student class and able to perform action 
+        ///  Output: Both object are same (true)
+        /// </summary>
+        Student student1 = Student.Instance;  // creating new object referring to same class 
+
+        if(student == student1)  // comaparing the both instances
+        {
+            Console.WriteLine("Both objects are same");   
+        }
+ 
+        student1.UpdateDetails("Ravi", 30, "A");  // updating the grade of student class using student1 object
+
```

Comment:
Good examples of use, but I think you could rewrite to make it a little easier to understand. In particular, the data are very similar (same name and age, and only a different grade), and I think if you presented the creation of `student` and `student1` together, with the demonstration from `DisplayDetails` immediately following that would be a clearer example. 

The boolean results are good too, but they could be presented at the bottom of the example as extra confirmation that the example works as expected.